python/cudf/cudf/core/column/column.py

```diff
@@ -1464,15 +1392,16 @@ def kernel_non_masked(values, results):
                 results[i] = core(values[i])
 
         kernel_non_masked.forall(len(column))(values, results)
-    # Output
-    return Buffer(results)
+
+    return results
 
 
 def _data_from_cuda_array_interface_desc(desc):
     ptr = desc["data"][0]
     nelem = desc["shape"][0]
     dtype = np.dtype(desc["typestr"])
 
+    # TODO: this can be done more efficiently
```

Comment:
Left comment above, but I think we should pass the obj to this function instead of just the dictionary.

Reply:
Done